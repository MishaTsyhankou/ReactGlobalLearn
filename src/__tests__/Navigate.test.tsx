import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Navigate from '../Components/Navigation/Navigation'





jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn,
}));

jest.mock('react-redux', () => ({
    useDispatch: jest.fn(() => jest.fn()),
    useSelector: jest.fn(() => ({ showByGenre: 'all' })),
  }));


describe('<Navigate />', () => {
  const TestGenreToggler = <Navigate  />;

  it('should match snapshot', () => {
    const { container } = render(<Navigate/>);

    expect(container).toMatchSnapshot();
  });

  it('All button should be active by default', () => {
    const { getByText } = render(TestGenreToggler);

    const allButton = getByText('All');

    expect(allButton.parentElement).toHaveClass('active');
  });

  
});