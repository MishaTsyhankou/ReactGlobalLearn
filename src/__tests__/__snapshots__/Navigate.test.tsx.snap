// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigate /> should match snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <ul
      class="menu"
    >
      <li
        class="item active"
        role="menuitem"
        tabindex="0"
      >
        <a
          class="link"
        >
          All
        </a>
      </li>
      <li
        class="item inActive"
        role="menuitem"
        tabindex="0"
      >
        <a
          class="link"
        >
          Documentary
        </a>
      </li>
      <li
        class="item inActive"
        role="menuitem"
        tabindex="0"
      >
        <a
          class="link"
        >
          Comedy
        </a>
      </li>
      <li
        class="item inActive"
        role="menuitem"
        tabindex="0"
      >
        <a
          class="link"
        >
          Horror
        </a>
      </li>
      <li
        class="item inActive"
        role="menuitem"
        tabindex="0"
      >
        <a
          class="link"
        >
          Crime
        </a>
      </li>
    </ul>
    <div>
      <div
        class="sortingDropdown"
      >
        <label
          class="label"
          for="sortingDropdown"
        >
          Sort by:
        </label>
        <div
          class="dropdown"
        >
          <select
            class="select"
            id="sortingDropdown"
          >
            <option
              value="release_date"
            >
              release date
            </option>
            <option
              value="budget"
            >
              budget
            </option>
            <option
              value="vote_average"
            >
              rating
            </option>
            <option
              value="vote_count"
            >
              popularity
            </option>
          </select>
          <img
            alt=""
            class="icon"
            height="10"
            src="[object Object]"
            width="22"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
